tools/mason/MasonInit.chpl

```diff
@@ -37,44 +37,71 @@ Initialises a library project in a project directory
 */
 proc masonInit(args) throws {
   try! {
-    var name = '';
+    var dirName = '';
     var show = false;
+    var packageName = '';
+    var countArgs = 2;
     for arg in args[2..] {
-      if arg == '-h' || arg == '--help' {
-        masonInitHelp();
-        exit();
-      }
-      else if arg == '--show' {
-        show = true;
-      }
-      else {
-        name = arg;
+      countArgs += 1;
+      select (arg) {
+        when '-h' {
+          masonNewHelp();
+          exit();
+        }
+        when '--help' {
+          masonNewHelp();
+          exit();
+        }
+        when '--show' {
+          show = true;
+        }
+        when '--name' {
+          packageName = args[countArgs];
+          dirName = args[2];
+        }
+        otherwise {
+          if arg.startsWith('--name') {
+            var res = arg.split("=");
+            packageName = res[2];
+            dirName = args[2];
+          }
+          if packageName.length == 0 then {
```

Comment:
length is being deprecated soon:

```suggestion
          if packageName.size == 0 then {
```

Reply:
This check depends on previous flags, which results in a flag order dependence, i.e. will these produce the same result?

```
mason init --name=foo bar
mason init bar --name=foo
```

I would suggest having `--name` set only the `packageName`, and this arg set only the `dirName`. After arg parsing is complete, you can set the `dirName` if it is not set, and then set the `packageName` it is not set.

Reply:
Same goes for `MasonNew`

Reply:
>> mason init --name=foo bar

I don't think an user would actually use the flag like this.
A new user would typically use `mason init illegal-name`. Then, after seeing the error message, he would follow correct syntax `mason init illegal-name --name legalName`.
I don't think an experienced user would use the flag before, unless deliberately. 
If this is not absolutely important, let's keep it as is.

Reply:
I agree that it's an odd usage, but I think a command line tool's arg parser should be robust enough to handle the arguments in any order (beyond subcommands, which require a specific order, e.g. `mason init` and not `mason --flag init`. 

I am not sure what the expected behavior of `mason init --name=foo bar` would be otherwise.